### ISSUE TITLE
Crunchyroll is broken 

### ISSUE THREAD

**Issue by fe80Grau** [fe80Grau/ytdlp2STRM#52]
https://github.com/yt-dlp/yt-dlp/issues/10207
It looks like Crunchyroll is already applying DRM to Switch endpoints. Let's wait and see if the yt-dlp team finds a solution, but I think the dark ages are just beginning.

**Comment by DarkYamik** [comment]
Thanks for your work, it does seem like the dark ages will begin 😭

**Comment by fe80Grau** [comment]
Easy things are not forever, however I shed a little light, soon I will integrate another application that with the CDM keys allows you to decrypt crunchyroll's DRM. I will also post a guide here on how to obtain these keya.

**Comment by sergiohzph** [comment]
I don't think it is legal in Spain what you intend to do, as ‘piracy’ as such is not allowed, and by integrating a functionality to get the L3 CDM keys you would be violating a lot of intellectual property. Many Crunchyroll downloader developers had to shelve their project because being from Europe they could not make a widevine decryption module due to the legal consequences.

**Comment by fe80Grau** [comment]
I totally agree and appreciate your warning, but I will never do anything illegal because ytdl2strm will never be able to dump CDMs. What I can do is code the necessary functions that allow it to be integrated with tools that do allow the download of said content if the CDMs are indicated. Dumping CDMs is a parallel process that can be easily found. But I repeat, ytdlp2strm will never do that function, it will only be a launcher, when this is ready it will obviously come with its respective disclaimer and legal notes to avoid problems.

**Comment by haraldmbs** [comment]
Appologeeze if you are already aware, but it seems this project does something similar to what you were describing: https://github.com/Crunchy-DL/Crunchy-Downloader

Maybe it can help?
If not sorry for wasting your time.
